src/cursor.c

```diff
@@ -206,16 +206,14 @@ Cursor *Cursors_Reserve(CursorList *cl, StrongRef global_spec_ref, unsigned inte
   // If the cursor should be associated with a spec,
   // we assume that global_spec_ref points to a valid spec, else the function returns NULL.
   IndexSpec *spec = StrongRef_Get(global_spec_ref);
-
   // If we are in a coordinator ctx, the spec is NULL
-  if (spec && (spec->activeCursors >= spec->cursorsCap)) {
+  if (spec && spec->activeCursors >= RSGlobalConfig.indexCursorLimit) {
     /** Collect idle cursors now */
     Cursors_GCInternal(cl, 0);
-    if (spec->activeCursors >= spec->cursorsCap) {
-      QueryError_SetError(status, QUERY_ELIMIT, "Too many cursors allocated for index");
+    if (spec->activeCursors >= RSGlobalConfig.indexCursorLimit) {
+      QueryError_SetErrorFmt(status, QUERY_ELIMIT, "CURSOR_LIMIT was reached for index, limit of %lld reached", RSGlobalConfig.indexCursorLimit);
```

Comment:
consider improving the grammar of this error

Reply:
```suggestion
      QueryError_SetErrorFmt(status, QUERY_ELIMIT, "CURSOR_LIMIT of %lld was reached for an index", RSGlobalConfig.indexCursorLimit);
```
?